webapp/src/components/user_attribute/index.ts

```diff
@@ -0,0 +1,50 @@
+import {connect} from 'react-redux';
+import {AnyAction, Dispatch, bindActionCreators} from 'redux';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {getGitlabUser} from '../../actions';
+
+import {GlobalState, pluginStateKey} from 'src/types/store';
+
+import UserAttribute from './user_attribute';
+
+interface UserAttributeStateProps {
+    id: string,
+    username: string,
+    gitlabURL: string,
+}
+
+interface UserAttributeDispatchProps {
+    actions: {
+        getGitlabUser:(userID: string) => (dispatch: Dispatch<AnyAction>, getState: () => GlobalState) => Promise<any>,
```

Comment:
I think `Promise<any>` can defeat the purpose of using typescript. Can we have the function we're calling here have a proper typing in `actions.ts`, as well as the other exported functions there? We want to set contributors up for success here as much as we can with types

We can then type this as

```suggestion
        getGitlabUser: typeof getGitlabUser,
```